File: ruby/rbfind/lib/rbfind/findoption.rb
# frozen_string_literal: true

module RbFind

  # FindOption - encapsulates a CLI find option
  class FindOption
    attr_reader :short_arg
    attr_reader :long_arg
    attr_reader :desc
    attr_reader :func

    def initialize(short_arg, long_arg, desc, func)
      @short_arg = short_arg
      @long_arg = long_arg
      @desc = desc
      @func = func
    end

    def sortarg
      if !@short_arg.nil? && !@short_arg.empty?
        @short_arg.downcase + 'a' + @long_arg.downcase
      else
        @long_arg.downcase
      end
    end
  end
end

File: typescript/tsfind/src/findoption.ts
/*
 * findoption.js
 *
 * encapsulates a find option
 */

'use strict';

import {ArgTokenType} from "./argtokentype";

export interface Option {
    shortArg: string;
    longArg: string;
    desc: string;
    argType: ArgTokenType;
    sortArg: string;
}

export class FindOption implements Option {
    shortArg: string;
    longArg: string;
    desc: string;
    argType: ArgTokenType;
    public sortArg: string;

    constructor(shortArg: string, longArg: string, desc: string, argType: ArgTokenType) {
        this.shortArg = shortArg;
        this.longArg = longArg;
        this.desc = desc;
        this.argType = argType;
        this.sortArg = this.getSortArg();
    }

    private getSortArg(): string {
        if (this.shortArg)
            return this.shortArg.toLowerCase() + 'a' + this.longArg.toLowerCase();
        return this.longArg.toLowerCase();
    }
}
